x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/core/tree/Node.java

```diff
@@ -431,20 +437,35 @@ final StringBuilder treeString(StringBuilder sb, int depth, BitSet hasParentPerD
     /**
      * Render the properties of this {@link Node} one by
      * one like {@code foo bar baz}. These go inside the
-     * {@code [} and {@code ]} of the output of {@link #treeString}.
+     * {@code [} and {@code ]} of the output of {@link #toString}.
      */
-    public String propertiesToString(boolean skipIfChild) {
+    protected String propertiesToString(boolean skipIfChild) {
+        return propertiesToString(skipIfChild, Node::toString, TO_STRING_MAX_PROP, TO_STRING_MAX_WIDTH);
+    }
+
+    /**
+     * Like the above, but ensures consistent tree rendering for golden tests. Overrides of this method should avoid adding information
+     * that can change between repeated test runs, such as object identities or pseudo-random generated names.
+     */
+    protected String goldenTestPropertiesToString(boolean skipIfChild) {
+        return propertiesToString(skipIfChild, Node::goldenTestToString, Integer.MAX_VALUE, Integer.MAX_VALUE);
+    }
+
+    private String propertiesToString(boolean skipIfChild, Function<Object, String> toStringHelper, int maxProperties, int maxWidth) {
         StringBuilder sb = new StringBuilder();
 
         @SuppressWarnings("HiddenField")
         List<?> children = children();
         // eliminate children (they are rendered as part of the tree)
-        int remainingProperties = TO_STRING_MAX_PROP;
-        int maxWidth = 0;
+        int remainingProperties = maxProperties;
+        int width = 0;
         boolean needsComma = false;
 
         List<Object> props = nodeProperties();
         for (Object prop : props) {
+            if (prop instanceof NameId) {
+                continue;
```

Comment:
Maybe it's irrelevant, but  instead of skipping, we could add a constant placeholder (eg. "$NameId$"), just to distinguish the cases where this element is present vs absent.